platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java

```diff
@@ -542,11 +544,31 @@ public CompassEngine getCompassEngine() {
   @Nullable
   @RequiresPermission(anyOf = {ACCESS_FINE_LOCATION, ACCESS_COARSE_LOCATION})
   public Location getLastKnownLocation() {
-    Location location = locationEngine != null ? locationEngine.getLastLocation() : null;
-    if (location == null) {
-      location = lastLocation;
+    if (locationEngine == null) {
+      return null;
+    }
+    // TODO: decide if we want to change signature of this method, for now make it work as is
+    final AtomicReference<LocationEngineResult> resultRef = new AtomicReference<>();
+    CountDownLatch latch = new CountDownLatch(1);
+    locationEngine.getLastLocation(new LocationEngineCallback<LocationEngineResult>() {
+      @Override
+      public void onSuccess(LocationEngineResult result) {
+        resultRef.set(result);
+        latch.countDown();
+      }
+
+      @Override
+      public void onFailure(@NonNull Exception exception) {
+        // TODO: log exception
+      }
+    });
+    try {
+      latch.await(2, TimeUnit.SECONDS);
+    } catch (InterruptedException ie) {
+      ie.printStackTrace();
     }
-    return location;
+    LocationEngineResult result = resultRef.get();
+    return result != null ? result.getLastLocation() : null;
```

Comment:
above codeblock seems to hold execution and freezing user interaction

Reply:
@tobrun indeed it does, this is not a recipe for integration, should probably expose a async shell method that wraps location engine call.